shared/output/gapic/templates/testing/lib/testing/routing_headers/service_explicit_headers/client.rb

```diff
@@ -0,0 +1,478 @@
+        #     - profiles/<profile_id>
+        #     - a legacy profile_id that can be any string
+        #   @param resource [::Testing::RoutingHeaders::RequestResource, ::Hash]
+        #
+        # @yield [response, operation] Access the result along with the RPC operation
+        # @yieldparam response [::Testing::RoutingHeaders::Response]
+        # @yieldparam operation [::GRPC::ActiveCall::Operation]
+        #
+        # @return [::Testing::RoutingHeaders::Response]
+        #
+        # @raise [::GRPC::BadStatus] if the RPC is aborted.
+        #
+        def plain request, options = nil
+          raise ::ArgumentError, "request must be provided" if request.nil?
+
+          request = ::Gapic::Protobuf.coerce request, to: ::Testing::RoutingHeaders::Request
+
+          # Converts hash and nil to an options object
+          options = ::Gapic::CallOptions.new(**options.to_h) if options.respond_to? :to_h
+
+          # Customize the options with defaults
+          metadata = @config.rpcs.plain.metadata.to_h
+
+          # Set x-goog-api-client and x-goog-user-project headers
+          metadata[:"x-goog-api-client"] ||= ::Gapic::Headers.x_goog_api_client \
+            lib_name: @config.lib_name, lib_version: @config.lib_version,
+            gapic_version: ::Testing::VERSION
+          metadata[:"x-goog-user-project"] = @quota_project_id if @quota_project_id
+
+          header_params = {}
+          if request.table_name
+            regex_match = %r{^(?<project_id>projects/[^/]+)/instances/[^/]+/tables/[^/]+/?$}.match request.table_name
+            if regex_match
+              header_params["project_id"] = regex_match["project_id".to_s]
+            end
+          end
+          if request.table_name
+            regex_match = %r{^projects/[^/]+/(?<instance_id>instances/[^/]+)/tables/[^/]+/?$}.match request.table_name
+            if regex_match
+              header_params["instance_id"] = regex_match["instance_id".to_s]
+            end
+          end
+          if request.table_name
+            regex_match = %r{^projects/[^/]+/instances/[^/]+/(?<table_id>tables/[^/]+)/?$}.match request.table_name
+            if regex_match
+              header_params["table_id"] = regex_match["table_id".to_s]
+            end
+          end
+          if request.table_name &&
+             %r{^projects/[^/]+/instances/[^/]+/tables/[^/]+/?$}.match?(request.table_name)
+            header_params["table_name"] = request.table_name
+          end
+          if request.table_name &&
+             %r{^projects/[^/]+/instances/[^/]+/tables/[^/]+/aliases(?:/.*)?$}.match?(request.table_name)
+            header_params["table_name"] = request.table_name
+          end
+          header_params["app_profile_id"] = request.app_profile_id if request.app_profile_id
+          header_params["app_profile_id_renamed"] = request.app_profile_id if request.app_profile_id
+          request_params_header = header_params.map { |k, v| "#{k}=#{v}" }.join("&")
```

Comment:
Will we need to escape any values here? e.g. base64-encoded values that include "=" characters, etc?